guides/STAKING-VALIDATOR.md

```diff
@@ -0,0 +1,308 @@
+    },
+    keyPairs: {
+      identity: validatorKey.json,
+      voteAccount: voteAccountKey.json,
+    },
+    flags: {
+      useSnapshotArchivesAtStartup: "when-newest",
+      fullRpcAPI: false,
+      rpcPort: 8899,
+      privateRPC: true,
+      onlyKnownRPC: true,
+      dynamicPortRange: "8002-8020",
+      gossipPort: 8001,
+      rpcBindAddress: "0.0.0.0",
+      walRecoveryMode: "skip_any_corrupted_record",
+      limitLedgerSize: 50000000,
+      blockProductionMethod: "central-scheduler",
+      fullSnapshotIntervalSlots: 1000,
+      noWaitForVoteToStartLeader: true,
+      noVoting: true,
+      entryPoint: networkInfo.entryPoint,
+      knownValidator: networkInfo.knownValidator,
+      expectedGenesisHash: networkInfo.genesisHash,
+    },
+  },
+  {
+    dependsOn: [instance],
+  },
+);
+
+// Expose information required to SSH to the validator host.
+export const nodes_name = ["instance"];
+export const nodes_public_ip = [instance.publicIp];
+export const nodes_private_key = [sshKey.privateKeyOpenssh];
+export const tuner_params = tunerParams;
+
+export const treasuryPublicKey = treasuryKey.publicKey;
+export const validatorPublicKey = validatorKey.publicKey;
+export const voteAccountPublicKey = voteAccountKey.publicKey;
+export const stakeAccountPublicKey = stakeAccountKey.publicKey;
+```
+
+## Treasury Funding Process
+
+When setting up your validator, funding the treasury keypair requires a two-stage deployment process:
+
+### Two-Stage Deployment
+
+1. **Stage 1: Deploy Only the Treasury Keypair**
+   
+   Initially modify your code to only create and export the treasury keypair:
+
+   ```typescript
+   // Create a treasury key for funding
+   const treasuryKey = new svmkit.KeyPair("treasury-key");
+
+   // Export the treasury public key
+   export const treasuryPublicKey = treasuryKey.publicKey;
+   ```
+
+   Deploy this simplified setup:
+   ```bash
+   pulumi up
+   ```
+
+   After deployment, retrieve the treasury public key:
+   ```bash
+   pulumi stack output treasuryPublicKey
+   ```
+
+   Fund this treasury address using an external wallet or faucet with enough SOL for both the validator identity transfer (100 SOL) and stake account creation (10 SOL).
+
+2. **Stage 2: Deploy the Complete Setup**
+   
+   Once the treasury and validator is funded, update your code to include all the transfer and stake account creation components, then run:
+   ```bash
+   pulumi up
+   ```
+
+This two-stage process is necessary because you need to know the treasury public key to fund it, but that key is only generated during the Pulumi deployment.
+
+2. **Network Configuration**: The code automatically uses the network configuration you already have set up, whether it's testnet, devnet, or mainnet.
+
+4. **Stake Amount**: The stake account is created with 0.3 SOL in this example. Adjust this based on your needs and available funds.
+
+## Deployment
+
+To deploy the enhanced configuration:
+
+```bash
+pulumi up
+```
```

Comment:
I think it reads better if the steps to pause and fund are interwolven with the code snipet steps. It should read as a step by step guide with what should be changed on on our non-voting example to get the validator voting and stakeable.